fineract-provider/src/main/java/org/apache/fineract/infrastructure/event/external/config/ExternalEventJMSConfiguration.java

```diff
@@ -60,4 +61,13 @@ public ActiveMQTopic activeMqTopic() {
     public ActiveMQQueue activeMqQueue() {
         return new ActiveMQQueue(fineractProperties.getEvents().getExternal().getProducer().getJms().getEventQueueName());
     }
+
+    @Bean("externalEventJmsProducerExecutor")
+    public ThreadPoolTaskExecutor externalEventJmsProducerExecutor() {
+        ThreadPoolTaskExecutor threadPoolTaskExecutor = new ThreadPoolTaskExecutor();
+        threadPoolTaskExecutor.setCorePoolSize(10);
```

Comment:
Would it make sense to move these into application.properties?

Reply:
It's just the core pool size, the threadpool can grow until the max pool size is reached (100 threads) but that's already a value nobody will utilize in my opinion. Essentially it'd mean 100 parallel producers.

Based on my initial measurement the sweet spot is around 30-40 ActiveMQ producers.